be/src/fs/fs_s3.cpp

```diff
@@ -149,39 +160,48 @@ S3ClientFactory::S3ClientPtr S3ClientFactory::new_client(const ClientConfigurati
 }
 
 static std::shared_ptr<Aws::S3::S3Client> new_s3client(const S3URI& uri, const FSOptions& opts) {
-    Aws::Client::ClientConfiguration config = S3ClientFactory::getClientConfig();
+    S3ClientFactory::ClientConfiguration config = S3ClientFactory::getClientConfig();
```

Comment:
`S3ClientFactory::getClientConfig()` will return a reference to static variable, it's not thread safe to modify it